stdlib/pathlib.pyi

```diff
@@ -232,8 +232,13 @@ class Path(PurePath):
     if sys.version_info >= (3, 9):
         def readlink(self) -> Self: ...
 
-    def rename(self, target: str | PurePath) -> Self: ...
-    def replace(self, target: str | PurePath) -> Self: ...
+    if sys.version_info >= (3, 10):
+        def rename(self, target: StrPath | PurePath) -> Self: ...
+        def replace(self, target: StrPath | PurePath) -> Self: ...
```

Comment:
`PurePath` is a sub-class of `PathLike[str]`:

```suggestion
        def rename(self, target: StrPath) -> Self: ...
        def replace(self, target: StrPath) -> Self: ...
```

Reply:
@srittau good catch, thank you for spotting it! Amended.